homeassistant/components/tessie/binary_sensor.py

```diff
@@ -0,0 +1,142 @@
+"""Binary Sensor platform for Tessie integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+@dataclass
+class TessieBinarySensorEntityDescription(BinarySensorEntityDescription):
```

Comment:
In a followup these should be changed due to a recent change https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes/

Reply:
This will come up a few times in the other platforms, but ill look at addressing them all once I fully understand the repercussions.